var/spack/repos/builtin/packages/py-setuptools/package.py

```diff
@@ -32,12 +32,28 @@ class PySetuptools(PythonPackage):
     homepage = "https://pypi.python.org/pypi/setuptools"
     url      = "https://pypi.python.org/packages/source/s/setuptools/setuptools-11.3.tar.gz"
 
+    import_modules = [
+        'pkg_resources', 'setuptools', 'pkg_resources.extern',
+        'pkg_resources._vendor', 'pkg_resources._vendor.packaging',
+        'setuptools.extern', 'setuptools.command'
+    ]
+
+    version('33.1.1', '7963d41d97b94e450e3f8a217be06ffe',
+            url="https://pypi.python.org/packages/dc/8c/7c9869454bdc53e72fb87ace63eac39336879eef6f2bf96e946edbf03e90/setuptools-33.1.1.zip")
     version('25.2.0', 'a0dbb65889c46214c691f6c516cf959c',
             url="https://pypi.python.org/packages/9f/32/81c324675725d78e7f6da777483a3453611a427db0145dfb878940469692/setuptools-25.2.0.tar.gz")
     version('20.7.0', '5d12b39bf3e75e80fdce54e44b255615')
     version('20.6.7', '45d6110f3ec14924e44c33411db64fe6')
-    version('20.5', 'fadc1e1123ddbe31006e5e43e927362b')
-    version('19.2', '78353b1f80375ca5e088f4b4627ffe03')
-    version('18.1', 'f72e87f34fbf07f299f6cb46256a0b06')
-    version('16.0', '0ace0b96233516fc5f7c857d086aa3ad')
+    version('20.5',   'fadc1e1123ddbe31006e5e43e927362b')
+    version('19.2',   '78353b1f80375ca5e088f4b4627ffe03')
+    version('18.1',   'f72e87f34fbf07f299f6cb46256a0b06')
+    version('16.0',   '0ace0b96233516fc5f7c857d086aa3ad')
     version('11.3.1', '01f69212e019a2420c1693fb43593930')
+
+    # Tests require:
+    # TODO: Add a 'test' deptype
+    # FIXME: All of these depend on setuptools, creating a dependency loop
+    # FIXME: Is there any way around this problem?
+    # depends_on('py-pytest-flake8', type='test')
+    # depends_on('pytest@2.8:', type='test')
+    # depends_on('py-mock', when='^python@:3.2', type='test')
```

Comment:
One constraint I want to put on the 'test' deptype is that both of the following commands would install the same package with the same hash:
```
spack install py-setuptools
spack install --run-tests py-setuptools
```
An unfortunate consequence of this is that we have no means of working around dependency loops. We can't do something similar to #2548 if we want to maintain this constraint, because the test dependency would have the same spec as the package that indirectly depends on it.

I don't think there is any good way around this problem. Let me now if you can think of anything better though.